attribute-utils/src/main/java/org/hypertrace/attribute/db/DbAttributeUtils.java

```diff
@@ -0,0 +1,156 @@
+    } else if (SpanAttributeUtils.containsAttributeKey(event, OTEL_DB_SYSTEM)) {
+      if (!OTEL_MONGO_DB_SYSTEM_VALUE.equals(event.getAttributes().getAttributeMap().get(
+          OTEL_DB_SYSTEM).getValue())) {
+        return Optional.empty();
+      }
+      return getSqlURIForOtelFormat(event);
+    }
+
+    return Optional.empty();
+  }
+
+  public static List<String> getTagsForMongoOperation() {
+    return Lists.newArrayList(Sets.newHashSet(OTHER_MONGO_OPERATION, OTEL_DB_OPERATION));
+  }
+
+  public static List<String> getTagsForMongoNamespace() {
+    return Lists.newArrayList(Sets.newHashSet(OTEL_MONGO_COLLECTION, OTHER_MONGO_NAMESPACE));
+  }
+
+  public static Optional<String> getRedisURI(Event event) {
+    if (SpanAttributeUtils.containsAttributeKey(event, OTHER_REDIS_CONNECTION)) {
+      return Optional.of(SpanAttributeUtils.getStringAttribute(event, OTHER_REDIS_CONNECTION));
+    }
+    if (event.getAttributes().getAttributeMap().containsKey(OTEL_DB_SYSTEM)) {
+      if (!OTEL_REDIS_DB_SYSTEM_VALUE.equals(
+          event.getAttributes().getAttributeMap().get(OTEL_DB_SYSTEM).getValue())) {
+        return Optional.empty();
+      }
+      return getSqlURIForOtelFormat(event);
+    }
+    return Optional.empty();
+  }
+
+  public static boolean isSqlBackend(Event event) {
+    if (event.getEventName() != null
+        && event.getEventName().startsWith(JDBC_EVENT_PREFIX)
+        && SpanAttributeUtils.containsAttributeKey(event, SQL_URL)) {
+      return true;
+    }
+    return isSqlTypeBackendForOtelFormat(event);
+  }
+
+  public static boolean isSqlTypeBackendForOtelFormat(Event event) {
+    if (SpanAttributeUtils.containsAttributeKey(event, OTEL_DB_SYSTEM)) {
+      return Arrays
+          .stream(OTEL_SQL_DB_SYSTEM_VALUES)
+          .anyMatch(
+              v -> v.equals(
+                  event.getAttributes().getAttributeMap().get(OTEL_DB_SYSTEM).getValue()));
+    }
+    return false;
+  }
+
+  public static boolean isSqlTypeBackendForOtelFormat(Map<String, AttributeValue> attributeMap) {
+    if (attributeMap.containsKey(OTEL_DB_SYSTEM)) {
+      return Arrays
+          .stream(OTEL_SQL_DB_SYSTEM_VALUES)
+          .anyMatch(
+              v -> v.equals(
+                  attributeMap.get(OTEL_DB_SYSTEM).getValue()));
+    }
+    return false;
+  }
+
+  public static Optional<String> getSqlURI(Event event) {
+    if (SpanAttributeUtils.containsAttributeKey(event, SQL_URL)) {
+      return Optional.of(SpanAttributeUtils.getStringAttribute(event, SQL_URL));
+    }
+    return getSqlURIForOtelFormat(event);
+  }
+
+  public static Optional<String> getSqlURIForOtelFormat(Event event) {
+    return getSqlURIForOtelFormat(event.getAttributes().getAttributeMap());
+  }
+
+  public static Optional<String> getSqlURIForOtelFormat(
+      Map<String, AttributeValue> attributeMap) {
+    if ((attributeMap.containsKey(OTEL_NET_PEER_NAME)
```

Comment:
Can we use `SpanAttributeUtils` method to read value?

Reply:
SpanAttributeUtils doesn't have any method to read directly from map